tests/fetch/example_idb_store.cpp

```diff
@@ -0,0 +1,40 @@
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <emscripten/fetch.h>
+
+void success(emscripten_fetch_t *fetch)
+{
+  printf("IDB store succeeded.\n");
+  emscripten_fetch_close(fetch);
+}
+
+void failure(emscripten_fetch_t *fetch)
+{
+  printf("IDB store failed.\n");
+  emscripten_fetch_close(fetch);
+}
+
+void persistFileToIndexedDB(const char *outputFilename, uint8_t *data, size_t numBytes)
+{
+  emscripten_fetch_attr_t attr;
+  emscripten_fetch_attr_init(&attr);
+  strcpy(attr.requestMethod, "EM_IDB_STORE");
+  attr.attributes = EMSCRIPTEN_FETCH_REPLACE | EMSCRIPTEN_FETCH_PERSIST_FILE;
+  attr.requestData = (char *)data;
```

Comment:
Data from `requestData` is copied by `HEAP8.slice`. 
`emscripten_fetch_close` frees the memory shallowly, hence allocated memort in line 35 will leak. 
It's not a big deal here, but it's not documented who owns memory of `requestData` and if `emscripten_fetch_close` will free it. The same applies for code blocks for the documentation.